### DISCUSSION TITLE
webclient2连接报错

### DISCUSSION THREAD

**Discussion by Kbstsn** [lejianwen/rustdesk-api#75] [Q&A]
这是我的docker启动命令，目前是用的最新版本，使用webclient2的时候依旧提示Failed to connect to rendezvous server, wss://hserver..::21118是还需要SSL反代21118和21119吗？
docker run -d --name rustdesk-api
-e TZ=Asia/Shanghai
-e RUSTDESK_API_LANG=zh-CN
-e RUSTDESK_API_RUSTDESK_ID_SERVER=hserver.:21116
-e RUSTDESK_API_RUSTDESK_RELAY_SERVER=hserver.:21117
-e RUSTDESK_API_RUSTDESK_API_SERVER=hserver.
-e RUSTDESK_API_GORM_TYPE=mysql
-e RUSTDESK_API_MYSQL_USERNAME=.\
-e RUSTDESK_API_MYSQL_PASSWORD=.\
-e RUSTDESK_API_MYSQL_ADDR=******:3306
-e RUSTDESK_API_MYSQL_DBNAME=rustdesk
-e RUSTDESK_API_RUSTDESK_KEY=d8KnL27MgMic7tDob4PDk7deUFzz9EMEbDP72mI74l8=
lejianwen/rustdesk-api

**Comment by lejianwen** [comment]
将端口号去掉试试RUSTDESK_API_RUSTDESK_ID_SERVER
这里的

**Reply by Kbstsn** [reply]
> 将端口号动手尝试RUSTDESK_API_RUSTDESK_ID_SERVER 这里的

刚试完，不行

**Reply by lejianwen** [reply]
看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document

**Reply by wujianfeng0722** [reply]
> 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document

我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢

**Reply by Kbstsn** [reply]
> > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> 
> 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢

我在WEB客户端，看见设备都是掉线状态的
![image](https://github.com/user-attachments/assets/213fe765-7f04-4d49-bfbb-20139a72ea77)

**Reply by lejianwen** [reply]
> > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> 
> 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢

如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？

**Reply by lejianwen** [reply]
> > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > 
> > 
> > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> 
> 我在WEB客户端，看见设备都是掉线状态的 ![image](https://private-user-images.githubusercontent.com/70641113/394104597-213fe765-7f04-4d49-bfbb-20139a72ea77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTMwMTIsIm5iZiI6MTczMzgxMjcxMiwicGF0aCI6Ii83MDY0MTExMy8zOTQxMDQ1OTctMjEzZmU3NjUtN2YwNC00ZDQ5LWJmYmItMjAxMzlhNzJlYTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDA2MzgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0MzU4MDhjNGFlYzdkNDk0ZDE1MzUyZGRmNGYyMmI0YTE4ZDJhZGNkNjc0NjZiYTk0ZDYyMzM5MzE4YjAyNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.waxUp3HqKai6yiioVd-FPGmKyAWQYDc-Yt9ZY3m2gSE)

反向代理配置正确吗？

**Reply by Kbstsn** [reply]
> > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > 
> > > 
> > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > 
> > 
> > 我在WEB客户端，看见设备都是掉线状态的 ![image](https://private-user-images.githubusercontent.com/70641113/394104597-213fe765-7f04-4d49-bfbb-20139a72ea77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTMwMTIsIm5iZiI6MTczMzgxMjcxMiwicGF0aCI6Ii83MDY0MTExMy8zOTQxMDQ1OTctMjEzZmU3NjUtN2YwNC00ZDQ5LWJmYmItMjAxMzlhNzJlYTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDA2MzgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk0MzU4MDhjNGFlYzdkNDk0ZDE1MzUyZGRmNGYyMmI0YTE4ZDJhZGNkNjc0NjZiYTk0ZDYyMzM5MzE4YjAyNmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.waxUp3HqKai6yiioVd-FPGmKyAWQYDc-Yt9ZY3m2gSE)
> 
> 反向代理配置正确吗？

我还没试过反代21118和21119这两个端口。晚点试试，哈哈

**Reply by wujianfeng0722** [reply]
> > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > 
> > 
> > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> 
> 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？

pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时

**Reply by lejianwen** [reply]
> > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > 
> > 
> > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> 
> 我在WEB客户端，看见设备都是掉线状态的 ![image](https://private-user-images.githubusercontent.com/70641113/394104597-213fe765-7f04-4d49-bfbb-20139a72ea77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MTQ1ODUsIm5iZiI6MTczMzgxNDI4NSwicGF0aCI6Ii83MDY0MTExMy8zOTQxMDQ1OTctMjEzZmU3NjUtN2YwNC00ZDQ5LWJmYmItMjAxMzlhNzJlYTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEwVDA3MDQ0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0ODExYzBmOWM4NDBkNWQ2MTQxNTE5MzcyZTU1OTI5MGY5OTIxMTgyM2UwNjFiNzg0NjI2YWEwZGI3MmQxNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RDyaS3qnQb1wP1ZJqKf59U6ekB3BrwgAbhpKI5U5_gc)

因为开源版本的server还没支持，你可以看看这个https://github.com/lejianwen/rustdesk-api/discussions/59 使用这个server解决这个问题

**Reply by lejianwen** [reply]
> > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > 
> > > 
> > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > 
> > 
> > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> 
> pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时

使用官方的会这样吗？http://rustdesk.com/web/

**Reply by wujianfeng0722** [reply]
> > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > 
> > > > 
> > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > 
> > > 
> > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > 
> > 
> > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> 
> 使用官方的会这样吗？http://rustdesk.com/web/

使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本

**Reply by lejianwen** [reply]
> > > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > > 
> > > > > 
> > > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > > 
> > > > 
> > > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > > 
> > > 
> > > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> > 
> > 
> > 使用官方的会这样吗？http://rustdesk.com/web/
> 
> 使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本

我更新了wiki，你看下https://github.com/lejianwen/rustdesk-api/wiki
我看官方webclient更新，我有时间更新一下，更新了我告诉你

**Reply by wujianfeng0722** [reply]
> > > > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > > > 
> > > > > > 
> > > > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > > > 
> > > > > 
> > > > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > > > 
> > > > 
> > > > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> > > 
> > > 
> > > 使用官方的会这样吗？http://rustdesk.com/web/
> > 
> > 
> > 使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本
> 
> 我更新了wiki，你看下https://github.com/lejianwen/rustdesk-api/wiki 我看官方webclient更新，我有时间更新一下，更新了我告诉你

好的，非常感谢，期待您的更新。

**Reply by lejianwen** [reply]
> > > > > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > > > > 
> > > > > > > 
> > > > > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > > > > 
> > > > > > 
> > > > > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > > > > 
> > > > > 
> > > > > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> > > > 
> > > > 
> > > > 使用官方的会这样吗？http://rustdesk.com/web/
> > > 
> > > 
> > > 使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本
> > 
> > 
> > 我更新了wiki，你看下https://github.com/lejianwen/rustdesk-api/wiki 我看官方webclient更新，我有时间更新一下，更新了我告诉你
> 
> 好的，非常感谢，期待您的更新。

我更新了，你可以试试

如果还不行
可以看看这个 https://github.com/lejianwen/rustdesk-api/wiki/HTTP-Reverse-Proxy
```
       # 这里不要填端口号
       - RUSTDESK_API_RUSTDESK_ID_SERVER=<server> 
       - RUSTDESK_API_RUSTDESK_RELAY_SERVER=<server> 
```

不过感觉也还是不行，因为你win10没问题。这就蛋疼了......

**Reply by wujianfeng0722** [reply]
> > > > > > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > > > > > 
> > > > > > > > 
> > > > > > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > > > > > 
> > > > > > > 
> > > > > > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > > > > > 
> > > > > > 
> > > > > > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> > > > > 
> > > > > 
> > > > > 使用官方的会这样吗？http://rustdesk.com/web/
> > > > 
> > > > 
> > > > 使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本
> > > 
> > > 
> > > 我更新了wiki，你看下https://github.com/lejianwen/rustdesk-api/wiki 我看官方webclient更新，我有时间更新一下，更新了我告诉你
> > 
> > 
> > 好的，非常感谢，期待您的更新。
> 
> 我更新了，你可以试试
> 
> 如果还不行 可以看看这个 https://github.com/lejianwen/rustdesk-api/wiki/HTTP-Reverse-Proxy
> 
> ```
>        # 这里不要填端口号
>        - RUSTDESK_API_RUSTDESK_ID_SERVER=<server> 
>        - RUSTDESK_API_RUSTDESK_RELAY_SERVER=<server> 
> ```
> 
> 不过感觉也还是不行，因为你win10没问题。这就蛋疼了......

webclient可以连上mac了，非常感谢大佬的更新和指导，webclient连mac延迟有点大，win反而挺顺畅，不过已经很满足了。

**Reply by lejianwen** [reply]
> > > > > > > > > > 看一下这个能解决不 https://github.com/lejianwen/rustdesk-api/wiki/Web-Client-V2-Preview-Document
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > 我也遇到同样的问题，也尝试这个方法，不可行，我是macos12.7，可以通过webclient2访问win10，但是mac就超时，请问还有其他解决方案吗？谢谢
> > > > > > > > 
> > > > > > > > 
> > > > > > > > 如果 `访问win10，但是mac就超时`，那可能是其他问题，用pc client 连mac 会超时吗？
> > > > > > > 
> > > > > > > 
> > > > > > > pc client正常连，只有webclient2连接mac会超时，我使用webclient1同样是超时
> > > > > > 
> > > > > > 
> > > > > > 使用官方的会这样吗？http://rustdesk.com/web/
> > > > > 
> > > > > 
> > > > > 使用官方的webclient也是正常连，docker环境如 Web-Client-V2-Preview-Document的http，还有哪些需要注意的配置吗？客户端，我使用您编译的1.3.2版本
> > > > 
> > > > 
> > > > 我更新了wiki，你看下https://github.com/lejianwen/rustdesk-api/wiki 我看官方webclient更新，我有时间更新一下，更新了我告诉你
> > > 
> > > 
> > > 好的，非常感谢，期待您的更新。
> > 
> > 
> > 我更新了，你可以试试
> > 如果还不行 可以看看这个 https://github.com/lejianwen/rustdesk-api/wiki/HTTP-Reverse-Proxy
> > ```
> >        # 这里不要填端口号
> >        - RUSTDESK_API_RUSTDESK_ID_SERVER=<server> 
> >        - RUSTDESK_API_RUSTDESK_RELAY_SERVER=<server> 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 不过感觉也还是不行，因为你win10没问题。这就蛋疼了......
> 
> webclient可以连上mac了，非常感谢大佬的更新和指导，webclient连mac延迟有点大，win反而挺顺畅，不过已经很满足了。

哈哈 那就好

**Reply by zisain** [reply]
我跟他遇到类似的问题,我使用域名http://rd.zin.top:58014访问就不行,一定会报错, Failed to connect to rendezvous server, ws://rd.zin.top:58018,但是如果我改成http://125.77.0.2:5814访问就不存在问题.匪夷所思.实际上我内部的dns配置 rd.zin.top就是解析到125.77.0.2
